### PR TITLE
Expand algorithm definition

### PR DESCRIPTION
**Pull request by sschaub**
I thought it would be helpful to expand the explanation of algorithms to include an example and discuss the relationship between algorithms, pseudocode, and programs.

### CODE REVIEW

File: _sources/GeneralIntro/Algorithms.rst [hunk 1]
```diff
@@ -11,15 +11,70 @@
...
+           program
+
 Algorithms
 ----------
+           
+If problem solving is a central part of computer science, then the solutions that you create through
+the problem solving process are also important.  In computer science, we refer to these solutions
+as **algorithms**.  An algorithm is a step by step list of instructions that if followed exactly will solve the problem under consideration.
+
+For example, an algorithm to sort a list of numbers in order from least to greatest might look like this:
```
Comment (bnmnetp) [inline _sources/GeneralIntro/Algorithms.rst:24]: the algorithm provided is for calculating the area of a circle, not for sorting...

File: _sources/GeneralIntro/Algorithms.rst [hunk 2]
```diff
@@ -11,15 +11,70 @@
...
+   :nocodelens:
+
+   radius = int(input("Enter the radius:"))
+   area = (radius * radius) * 3.1415
+   print("The area of a circle with radius", radius, "is:", area)
+
+A **program** is an algorithm expressed in a programming language. We might also say
+that a program is an *implementation* of an algorithm. In this example, both the
+algorithm and the program have three steps. The first step gets some input from
+the user; the second step performs a calculation using the information obtained
```
Comment (bnmnetp) [inline _sources/GeneralIntro/Algorithms.rst:66]: and transforms the input into something that the computer can do math with??

File: _sources/GeneralIntro/Algorithms.rst (hunk 1 above)
Comment (sschaub) [reply _sources/GeneralIntro/Algorithms.rst]: I've fixed this. Sorry about that...

File: _sources/GeneralIntro/Algorithms.rst (hunk 2 above)
Comment (sschaub) [reply _sources/GeneralIntro/Algorithms.rst]: I incorporated your suggestion.

**Pull request merged**